collector/internal/config.go

```diff
@@ -0,0 +1,95 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package internal // import "go.opentelemetry.io/ebpf-profiler/collector/internal"
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/tklauser/numcpus"
+	"go.opentelemetry.io/collector/component"
+
+	tracertypes "go.opentelemetry.io/ebpf-profiler/tracer/types"
+)
+
+const (
+	defaultProjectID = "1"
+	defaultHostID    = 0x1234
+)
+
+// Config represents the receiver config settings within the collector's config.yaml
+type Config struct {
```

Comment:
Maybe for another PR, but it'd be nice if we used a single configuration structure to cover both CLI and collector, that way the validation logic would also be shared.